_gtfobins/ssh.md

```diff
@@ -3,6 +3,20 @@ functions:
   execute-interactive:
     - description: Reconnecting may help bypassing restricted shells.
       code: ssh localhost $SHELL --noprofile --norc
+    - description: Spawn interactive shell through ProxyCommand option.
+      code: ssh -o ProxyCommand="/bin/bash -c 'exec 10<&0 11>&1 0<&2 1>&2; /bin/bash -i'" whatever.invalid
+  execute-non-interactive:
```

Comment:
You can drop `execute-non-interactive` if `execute-interactive` exists with the same technique.

Reply:
I was thinking about that too.
The `execute-interactive` uses file descriptor trickery which probably won't work in all (restricted) environments.
That's why I kept a simple `execute-non-interactive` version which basically will work in almost all instances.
What do you think?

Reply:
You have a point there, @cyrus-and what do you thinK?

Reply:
@roman-mueller can you provide a real-world example where such trickery doesn't work?

Reply:
In the `bash -r` restricted shell scenario we usually use as example, such as below, the file descriptor trickery works, i.e.

```
mkdir /tmp/fs
cp /bin/ls /usr/bin/ssh /tmp/fs/
cd /tmp/fs
PATH=/tmp/fs /bin/bash --norc --noprofile -r
ssh -o ProxyCommand="/bin/sh -c 'exec 10<&0 11>&1 0<&2 1>&2; /bin/sh -i'" x
```

For this reason I would just keep the `execute-interactive` one, it is a pretty neat example.